samples/trumpify/trumpify.py

```diff
@@ -0,0 +1,120 @@
+import os
+import sys
+import threading
+import time
+
+sys.path.append(os.path.abspath('../../'))
+
+from samples.trumpify.utils import TwitterUtils
+from daemo.client import DaemoClient
+
+CREDENTIALS_FILE = 'credentials.json'
+
+PROJECT_ID = -1
+
+INPUT_TWITTER_NAME = 'HillaryClinton'
+MINUTES = 60
+FETCH_INTERVAL_MIN = 5
+TWEET_COUNT = 10
+MONITOR_INTERVAL_MIN = 60
+
+twitter = TwitterUtils()
+daemo = DaemoClient(CREDENTIALS_FILE)
+
+
+def transform_new_tweets(twitter_name, count, interval):
+    """
+    Fetches "count" number of tweets from twitter feed of "twitter_name" after every "interval" seconds and posts them
+    to Daemo server for translation
+
+    :param twitter_name: twitter account to fetch tweets from
+    :param count: number of tweets to fetch
+    :param interval: period in seconds to repeat the process
+    """
+    while True:
+        messages = twitter.fetch_tweets(twitter_name=twitter_name, count=count, interval=interval)
+
+        if len(messages) > 0:
+            for message in messages:
+                translate_to_trump_version(message)
+        else:
+            print "@%s has not tweeted in the last %d minutes." % (twitter_name, interval)
+
+        time.sleep(interval)
+
+
+def translate_to_trump_version(message):
+    """
+    Create a Daemo task using "message" as data and pass it to workers for translation
+
+    :param message: tweet object to be used as data for daemo task
+    """
+    text = message.get('text')
+    id = message.get('id')
+
+    daemo.publish(project_id=PROJECT_ID, tasks=[{
+        "id": id,
+        "tweet": text
+    }], approve=approve_tweet, completed=post_to_twitter)
+
+
+def get_tweet_text(worker_response):
+    """
+    Filter out just the tweet text from a worker's complete submission
+
+    :param worker_response: submission made by a worker for a task
+    :return: actual tweet text
+    """
+    return worker_response.get('results')[0].get('result')
+
+
+def approve_tweet(worker_responses):
+    """
+    Verify each worker response if it meets the requirements
+
+    :param worker_responses: submission made by a worker for a task
+    :return: list of True/False
+    """
+    approvals = [len(get_tweet_text(response)) > 0 for response in worker_responses]
+    return approvals
+
+
+def post_to_twitter(worker_responses):
+    """
+    Post worker's response to twitter and add to monitoring list
+
+    :param worker_responses: submission made by a worker for a task
+    """
+    for worker_response in worker_responses:
+        twitter.post(worker_response)
+
+
+def rate_worker_responses(interval):
+    """
+    Use tweet's retweet count in first "interval" seconds as criterion for rating workers which is pushed
+    to Daemo's reputation system
+
+    :param interval: period in seconds to wait before checking retweet count for a tweet
+    """
+    while True:
+        tweet = twitter.monitor_next_tweet(interval)
```

Comment:
We are processing a queue of tweets which were posted by workers here. Also we use interval to compute how long it should "sleep" to fetch the retweet count.